services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/services/status_monitor/_monitor.py

```diff
@@ -31,6 +33,34 @@ async def _start_get_status_deferred(
     await DeferredGetStatus.start(node_id=node_id)
 
 
+def _can_be_removed(model: TrackedServiceModel) -> bool:
+
+    # requested **as** `STOPPED`
+    # service **reports** `IDLE`
+    if (
+        model.current_state == SchedulerServiceState.IDLE
+        and model.requested_state == UserRequestedState.STOPPED
+    ):
+        return True
+
+    # NOTE: currently dynamic-scheduler does nto automatically start a
+    # service reported who's requested_state is STARTED
+    # to avoid monitoring services which no longer exist,
+    # the service has to be removed.
+
+    # requested as `STARTED`
+    # service **reports** `IDLE` since `_REMOVE_AFTER_IDLE_FOR`
+    if (  # noqa: SIM103
+        model.current_state == SchedulerServiceState.IDLE
+        and model.requested_state == UserRequestedState.RUNNING
+        and arrow.utcnow().timestamp() - model.last_state_change
+        > _REMOVE_AFTER_IDLE_FOR.total_seconds()
+    ):
+        return True
+
+    return False
```

Comment:
TIP: 
```python
return (
        model.current_state == SchedulerServiceState.IDLE
        and model.requested_state == UserRequestedState.STOPPED
    ) or ( 
        model.current_state == SchedulerServiceState.IDLE
        and model.requested_state == UserRequestedState.RUNNING
        and arrow.utcnow().timestamp() - model.last_state_change
        > _REMOVE_AFTER_IDLE_FOR.total_seconds()
    )
```

Reply:
or logically equivalent (and shorter :-) )
```python
return return model.current_state == SchedulerServiceState.IDLE and (
    model.requested_state == UserRequestedState.STOPPED
    or (
        model.requested_state == UserRequestedState.RUNNING
        and arrow.utcnow().timestamp() - model.last_state_change > _REMOVE_AFTER_IDLE_FOR.total_seconds()
    )
)
```

Reply:
The point was clarity. I chose to the verbose longer version with notes to make it more clear for future reading.